frontend/packages/knative-plugin/src/components/services/ServiceDetailsPage.tsx

```diff
@@ -1,28 +1,106 @@
+import {
+  ContainerTable,
+  DetailsItem,
+  ExternalLink,
+  ResourceSummary,
+  SectionHeading,
+  navFactory,
+} from '@console/internal/components/utils';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
 import { K8sResourceKind, referenceForModel, K8sKind } from '@console/internal/module/k8s';
 import {
   ActionMenu,
   ActionMenuVariant,
   ActionServiceProvider,
   useTabbedTableBreadcrumbsFor,
 } from '@console/shared';
+import { RevisionModel } from '../../models';
 import { isServerlessFunction } from '../../topology/knative-topology-utils';
+import { RevisionKind, ServiceKind, ServiceTypeValue } from '../../types';
 import { serverlessTab } from '../../utils/serverless-tab-utils';
-import ServerlessFunctionType from '../overview/ServerlessFunctionType';
+import { KnativeServiceTypeContext } from '../functions/ServiceTypeContext';
+import RevisionsOverviewList from '../overview/RevisionsOverviewList';
+
+const ServiceDetails: React.FC<{ obj: ServiceKind }> = ({ obj }) => {
+  const { t } = useTranslation();
+  const [revisions, revisionLoaded, revisionErrorLoad] = useK8sWatchResource<RevisionKind[]>({
+    kind: referenceForModel(RevisionModel),
+    namespace: obj.metadata.namespace,
+    isList: true,
+  });
+
+  const revisionsForService =
+    revisionLoaded &&
+    !revisionErrorLoad &&
+    revisions.filter(
+      (revision) =>
+        _.get(revision.metadata, `labels["serving.knative.dev/service"]`) === obj.metadata.name,
+    );
+  return (
+    <>
+      <div className="co-m-pane__body">
+        <SectionHeading text={t('knative-plugin~Service details')} />
+        <div className="row">
+          <div className="col-md-6">
+            <ResourceSummary
+              resource={obj}
+              podSelector="spec.podSelector"
+              showNodeSelector={false}
+            />
+          </div>
+          <div className="col-md-6">
+            {isServerlessFunction(obj) && (
+              <DetailsItem label={t('knative-plugin~Type')} obj={obj}>
+                {t('knative-plugin~Function')}
+              </DetailsItem>
+            )}
+            {obj?.status?.url && (
+              <DetailsItem label={t('knative-plugin~URL')} obj={obj} path="status.url">
+                <ExternalLink
+                  href={obj.status.url}
+                  additionalClassName="co-external-link--block"
+                  text={obj.status.url}
+                />
+              </DetailsItem>
+            )}
+            {revisionsForService && revisionLoaded && !revisionErrorLoad && (
+              <DetailsItem label={t('knative-plugin~Revisions')} obj={obj} path="status.traffic">
+                <RevisionsOverviewList
+                  revisions={revisionsForService}
+                  service={obj}
+                  hideSectionHeading
+                />
+              </DetailsItem>
+            )}
+          </div>
```

Comment:
The `DetailsItem` component renders `dt` and `dd`, these HTML elements must be wrapped in a `dl`:

```tsx
          <div className="col-md-6">
            <dl>
            {isServerlessFunction(obj) && (
              <DetailsItem label={t('knative-plugin~Type')} obj={obj}>
                {t('knative-plugin~Function')}
              </DetailsItem>
            )}
...
```